java-checks/src/test/files/checks/AnonymousClassShouldBeLambdaCheck_no_version.java

```diff
@@ -92,6 +102,20 @@ public String handle() {
         return A.this.toStr();
       }
     };
+   
+    new B() {// Compliant
+      @Override
+      public void bar() {
+      }
+    };
+  
+    new C() { // Noncompliant
+      @Override
+      public void foo() {
+      }
+    };
+   
+    new D() {};  // Compliant
```

Comment:
Where is `D` defined? if you want to explicit the case of a non-resovled class type, make it explicit using a proper name, such as ` new MyUnknownType() {};`